File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        n = len(s)
        track = defaultdict(int)
        left = considerCharCount = res = 0

        for right in range(n):
            track[s[right]] += 1
            considerCharCount = max(track[s[right]], considerCharCount)
            if right - left + 1 - considerCharCount > k:
                track[s[left]] -= 1
                if not track[s[left]]:
                    track.pop(s[left])
                left += 1

            res = max(res, right - left + 1)
        
        return res

File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.go
func characterReplacement(s string, k int) int {

    n := len(s)
    track := make(map[byte]int, n)

    currentConsider := 0

    left := 0

    maxLength := 0
    for right := 0; right < n; right++ {
        track[s[right]]++

        currentConsider = max(track[s[right]], currentConsider)
        
        if right - left + 1 - currentConsider > k {
            track[s[left]]--
            left++
        }

        maxLength = max(maxLength, right - left + 1)
    }

    return maxLength
}